### PR TITLE
[AN-4391] Disable tracking

### PR DESCRIPTION
**Pull request by hakonbo**
#### APK
[Download build #7349](http://192.168.10.18:8080/job/Pull%20Request%20Builder/7349/artifact/build/artifact/wire-dev-PR44-7349.apk)
[Download build #7351](http://192.168.10.18:8080/job/Pull%20Request%20Builder/7351/artifact/build/artifact/wire-dev-PR44-7351.apk)

### CODE REVIEW

File: app/src/main/java/com/waz/zclient/MainActivity.java [hunk 1]
```diff
@@ -220,9 +222,14 @@ protected void onResume() {
...
-                                             getControllerFactory().getUserPreferencesController().getDeviceId(),
-                                             getControllerFactory().getTrackingController());
+        final boolean trackingEnabled = getSharedPreferences(UserPreferencesController.USER_PREFS_TAG, Context.MODE_PRIVATE)
+            .getBoolean(getString(R.string.pref_advanced_analytics_enabled_key), true);
+
+        if (trackingEnabled) {
+            HockeyCrashReporting.checkForCrashes(getApplicationContext(),
+                                                 getControllerFactory().getUserPreferencesController().getDeviceId(),
+                                                 getControllerFactory().getTrackingController());
+        }
```
Comment (zbsz) [inline app/src/main/java/com/waz/zclient/MainActivity.java]: So when it's disabled then we won't upload crashes, but will ever delete the generated files? We may end up with multiple crash reports in files dir that are never deleted.

Comment (hakonbo) [inline app/src/main/java/com/waz/zclient/MainActivity.java]: We may get some exceptions there during current app session, but from next restart the UncaughtExceptionHandler won't be registered and no more exceptions stored.

Comment (hakonbo) [inline app/src/main/java/com/waz/zclient/MainActivity.java]: Ah, I guess you might refer to the manually saved exceptions? I'll have another look and see if we can address that too.

Comment (zbsz) [inline app/src/main/java/com/waz/zclient/MainActivity.java]: SE also saves some exceptions using `ExceptionHandler.saveException`.
We also have `HockeyCrashReporter` class, it was supposed to be used to upload crashes, but seems it's not used

Comment (hakonbo) [inline app/src/main/java/com/waz/zclient/MainActivity.java]: Good point. I'll add a delete call for every startup.

**Comment by hakonbo** [comment]
@zbsz Added stack trace purge

**Comment by zbsz** [comment]
Should we also add a check in `NativeCrashReporter`?

**Comment by hakonbo** [comment]
Not sure I follow regarding NativeCrashReporter. But if you suggest we should delete the native crash dumps created by breakpad, I agree. Will add a delete method to NativeCrashManager. Or were you thinking of something else?

**Comment by zbsz** [comment]
I was thinking about making sure that it doesn't upload when tracking is disabled.
Didn't notice that this is only done inside of `checkFroCrashes`, so its fine.
👍

**Comment by hakonbo** [comment]
Cool, pushed the native crash deletion stuff now...

**Pull request merged**